spock-core/src/main/java/org/spockframework/runtime/BaseSpecRunner.java

```diff
@@ -176,28 +173,38 @@ public Object invoke(Object target, Object... arguments) {
   public void doRunSetupSpec(SpecInfo spec) {
     runSetupSpec(spec.getSuperSpec());
     for (MethodInfo method : spec.getSetupSpecMethods()) {
-      if (runStatus != OK) return;
-      invoke(currentInstance, method);
+      if (runStatus.get() != OK) return;
+      invoke(NO_CURRENT_FEATURE, NO_CURRENT_ITERATION, sharedInstance, sharedInstance, method);
     }
   }
 
   private void runFeatures() {
-    for (FeatureInfo feature : spec.getAllFeaturesInExecutionOrder()) {
+    for (final FeatureInfo feature : spec.getAllFeaturesInExecutionOrder()) {
       if (resetStatus(FEATURE) != OK) return;
-      currentFeature = feature;
-      runFeature();
-      currentFeature = null;
+
+      if (feature.isReportIterations()) {
```

Comment:
Why are you using `isReportIterations()` instead of  `isParameterized()`? `doRunFeature` only tests for `isParameterized()` to invoke the `runParameterizedFeature` method.